experimental/features_example_test.go

```diff
@@ -0,0 +1,80 @@
+package experimental_test
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"log"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/experimental"
+	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+)
+
+// pthreadWasm was generated by the following:
+//
+//	docker run -it --rm -v `pwd`/testdata:/workspace ghcr.io/webassembly/wasi-sdk:wasi-sdk-20 sh -c '$CC -o /workspace/pthread.wasm /workspace/pthread.c --target=wasm32-wasi-threads --sysroot=/wasi-sysroot -pthread -mexec-model=reactor -Wl,--export=run -Wl,--export=get'
+//
+//go:embed testdata/pthread.wasm
+var pthreadWasm []byte
+
+// This shows how to use a WebAssembly module using pthread, compiled by wasi-sdk.
+func Example_pthread() {
+	// Use a default context
+	ctx := context.Background()
+
+	// Threads support is currently only supported with interpreter, so the config
+	// must explicitly specify it.
+	cfg := wazero.NewRuntimeConfigInterpreter()
+
+	// Threads support must be enabled explicitly in addition to standard V2 features.
+	cfg = cfg.WithCoreFeatures(api.CoreFeaturesV2 | experimental.CoreFeaturesThreads)
+
+	r := wazero.NewRuntimeWithConfig(ctx, cfg)
+	defer r.Close(ctx)
+
+	// Because we are using wasi-sdk to compile the guest, we must initialize WASI.
+	wasi_snapshot_preview1.MustInstantiate(ctx, r)
+
+	mod, err := r.Instantiate(ctx, pthreadWasm)
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	// Channel to synchronize start of goroutines before running.
+	startCh := make(chan struct{})
+	// Channel to synchronize end of goroutines.
+	endCh := make(chan struct{})
+
+	// We start up 8 goroutines and run for 6000 iterations each. The count should reach
```

Comment:
is it possible to use hammer.NewHammer instead? It would reduce the sorts of code that launch test goroutines.

Reply:
I thought it'd be better to avoid an internal method in an example, but if it makes sense then I'll switch to it